Test for 1-my_list.py
-------------------------------------------

import case
>>> MyList = __import__('1-my_list').MyList
>>>


Normal case 
>>> my_list = MyList()
>>> my_list.append(1)
>>> my_list.append(4)
>>> my_list.append(2)
>>> my_list.append(3)
>>> my_list.append(5)
>>> print(my_list)
[1, 4, 2, 3, 5]
>>> my_list.print_sorted()
[1, 2, 3, 4, 5]
>>> print(my_list)
[1, 4, 2, 3, 5]
>>>

An exception where the instance name is not defined
>>> MyList = __import__('1-my_list').MyList
>>> my_list2.print_sorted()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'my_list2' is not defined
>>>


Passed a list as an argument to the class
>>> a = [1, 9, 4]
>>> my_list = MyList(a)
>>> my_list.print_sorted()
[1, 4, 9]
>>>

Accessing an empty list
>>> a = []
>>> my_list = MyList(a)
>>> my_list.print_sorted()
[]
>>>

Handling a unique number
>>> my_list = MyList()
>>> my_list.append(1)
>>> my_list.print_sorted()
[1]
>>> print(my_list)
[1]
>>>


Printing a list of identical number
>>> my_list2 = MyList()
>>> my_list2.append(3)
>>> my_list2.append(3)
>>> my_list2.append(3)
>>> my_list2.print_sorted()
[3, 3, 3]
>>>


Accessing the parent object (list class)
>>> my_list = [1, 2, 3]
>>> my_list.print_sorted()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list' object has no attribute 'print_sorted'
>>>

Handling negative cases
>>> my_list = MyList()
>>> my_list.append(1)
>>> my_list.append(-1)
>>> my_list.append(51)
>>> my_list.append(0)
>>> my_list.append(-15)
>>> my_list.print_sorted()
[-15, -1, 0, 1, 51]
>>>
